bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```diff
@@ -130,8 +133,15 @@ public MavenWorkspaceRepository() {
 
 	@Activate
 	void activate() {
-		process();
 		list = memoize(this::list0);
+		promiseFactory.submit(this::process)
+			.thenAccept(changed -> {
+				if (changed) {
+					list = memoize(this::list0);
+					unchecked(() -> Central.refreshPlugins());
```

Comment:
You don't need `unchecked` here since `thenAccept` allows an exception to be thrown.

Reply:
ah, good one. I missed that.

Reply:
pushed